### PR TITLE
[feature][exiftools] Implement tags reading, allow for individual tag value read

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR finishes the tags reading exif conversion implementation, and adds a function to read an individual tag to the QgsExifTools class. 

E.g.: `QgsExifTools.readTag('/my/photo/0997.JPG'), 'Exif.Image.DateTime')`

Implementation note: known exif date/time/datetime are converted to Q{Date,Time,DateTime} objects. This makes working with tags much nicer, and will come in handy in expression contexts too.

### CODE REVIEW

**Comment by nicogodet** [comment]
Hi @nirvn,
Is it planned to implement this in Picture element in layouts ?
It would fix https://github.com/qgis/QGIS/issues/38611

**Comment by nirvn** [comment]
@nicogodet , you'll be able to use layout item rotation's data defined button and build an expression based on the exif orientation value yes.

That should be a nice workaround. A nice feature addition would be for a checkbox to activate auto rotation whenever an exif orientation value is present. That's not on my personal to do list, so unless someone adds a bounty, the workaround will do ;)

**Comment by nicogodet** [comment]
This feature request https://github.com/qgis/QGIS/issues/42444 could also benefit of your work

File: src/core/raster/qgsexiftools.cpp [hunk 1]
```diff
@@ -18,80 +18,147 @@
...
-        break;
-
-      case Exiv2::unsignedRational:
-      case Exiv2::signedRational:
       case Exiv2::unsignedByte:
       case Exiv2::signedByte:
-      case Exiv2::undefined:
       case Exiv2::tiffIfd:
+      case Exiv2::tiffIfd8:
+        val = QVariant::fromValue( static_cast< int >( it->toLong() ) );
```
Comment (m-kuhn) [inline src/core/raster/qgsexiftools.cpp:111]: Is the cast here required?

**Review by m-kuhn — APPROVED** [review]
:tada:

File: src/core/raster/qgsexiftools.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/raster/qgsexiftools.cpp:111]: It helps providing a more meaningful value when someone checks against the resulting QVariant's type().

**Comment by Houska1** [comment]
Hurrah! Any chance of accessing via QGIS expression function wrappers?

**Comment by nirvn** [comment]
@Houska1 , implemented expression function #44081 .

**Pull request merged**